From e1f965925af0a6a841d8b0646903fa95cfb8633f Mon Sep 17 00:00:00 2001
From: Nadav Hugim <nhugim@habana.ai>
Date: Fri, 03 Dec 2021 09:55:28 +0200
Subject: [PATCH] [SW-60390] fit: remove unused security function

this check moved into Guadi security in previous
patch and now can be remove

Change-Id: Ice4916283013069316a6027cc785e1c2ced9e655
Signed-off-by: Nadav Hugim <nhugim@habana.ai>
---

diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index be998a5..ffb11ef 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -394,48 +394,6 @@
 	return fdt_subnode_offset(fit, images, FIT_PUB_KEY_PROP);
 }
 
-#ifdef FAKE_SECURITY
-__weak void preboot_security_sp_sram_write_key_hash(u8 *key, size_t len) {}
-
-static int fit_pub_key_value_node_get(const void *fit)
-{
-	int conf = 0;
-	int hash_len = 0;
-	const u8 *hash_value = NULL;
-
-	conf = fit_pub_key_node_get(fit, conf);
-	if (conf < 0) {
-		preboot_puts("Wrong FIT format missing pubkey\n");
-		return 1;
-	}
-
-	conf =  fdt_subnode_offset(fit, conf, FIT_PUB_KEY_HASH_PROP);
-	if (conf < 0) {
-		preboot_puts("Wrong FIT format missing pubkey hash\n");
-		return 1;
-	}
-
-	hash_value = fdt_getprop(fit, conf, FIT_VALUE_PROP, &hash_len);
-	if (conf < 0) {
-		preboot_puts("Cannot find hash node\n");
-		return 1;
-	}
-
-	/*
-	 * Pass public key value to u-boot.
-	 * u-boot uses this information to create fake OTP
-	 * used in fake security mode.
-	 */
-	preboot_security_sp_sram_write_key_hash((u8 *)hash_value, hash_len);
-	return 0;
-}
-#else
-static int fit_pub_key_value_node_get(const void *fit)
-{
-	return 0;
-}
-#endif
-
 static int fit_pub_key_ext_node_get(const void *fit, int conf)
 {
 	int images;
@@ -811,9 +769,6 @@
 		return FIT_LOAD_CORRUPTED;
 
 	if (fit_verify_image(fit, fit_size, conf, res))
-		return FIT_LOAD_SEC_FAILED;
-
-	if (fit_pub_key_value_node_get(fit))
 		return FIT_LOAD_SEC_FAILED;
 
 	num_images = fdt_stringlist_count(fit, conf, FIT_LOADABLE_PROP);
